client/src/components/Tea/TeaDetails/TeaDetailsContainer.jsx

```diff
@@ -2,31 +2,56 @@
 import React, { Component } from "react";
 import { withRouter } from "react-router-dom";
 import { connect } from "react-redux";
-import { deleteTea } from "../../../actions/teaActions";
+import { deleteTea, getTeas } from "../../../actions/teaActions";
 import { TeaDetails } from "./TeaDetails";
 import { editTeaFlash } from "../../../actions/flashActions";
 
 class TeaDetailsContainer extends Component {
+  state = {
+    teaId: "",
+    name: "",
+    brand: "",
+    teaType: "",
+    servings: ""
+  };
+
   clickHandler = (e, status) => {
     this.props.updateFlash(status);
   };
 
+  componentDidMount() {
+    this.props.getTeaList(this.props.userID);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (this.props.tea.id !== prevState.teaId) {
+      this.setState({
+        teaId: this.props.tea.id,
+        name: this.props.tea.name,
+        brand: this.props.tea.brand,
+        teaType: this.props.tea.teaType,
+        servings: this.props.tea.servings
+      });
+    }
+  }
+
   render() {
     return (
       <TeaDetails
-        tea={this.props.tea}
+        tea={this.state}
```

Comment:
Can you explain what benefit you're getting by destructuring `props.tea` into the local state of this component? Seems unnecessary.

Reply:
on page refresh (user initiated, for example) the details view did not have the props it needed. there may have been some other logic needed to handle this better.

Reply:
That doesn't explain why it would be any different in the component's state. That would be blank upon page refresh as well.